addons/sale_project/models/project.py

```diff
@@ -370,20 +372,35 @@ def _get_profitability_sale_order_items_domain(self, domain=None):
     def _get_revenues_items_from_sol(self, domain=None, with_action=True):
         sale_line_read_group = self.env['sale.order.line'].sudo()._read_group(
             self._get_profitability_sale_order_items_domain(domain),
-            ['product_id'],
+            ['product_id', 'currency_id'],
             ['id:array_agg', 'untaxed_amount_to_invoice:sum', 'untaxed_amount_invoiced:sum'],
         )
         display_sol_action = with_action and len(self) == 1 and self.user_has_groups('sales_team.group_sale_salesman')
         revenues_dict = {}
         total_to_invoice = total_invoiced = 0.0
         if sale_line_read_group:
-            sols_per_product = {
-                product.id: (
-                    untaxed_amount_to_invoice,
-                    untaxed_amount_invoiced,
-                    ids,
-                ) for product, ids, untaxed_amount_to_invoice, untaxed_amount_invoiced in sale_line_read_group
-            }
+            dict_rate_per_currency = {}
+            today = datetime.now()
+            convert_company = self.company_id or self.env.company
+            sols_per_product = defaultdict(lambda: [0.0, 0.0, []])
+            for res in sale_line_read_group:
+                currency_id = res[1].id
+                if currency_id == convert_company.currency_id.id:
+                    current_list = sols_per_product[res[0].id]
```

Comment:
The current list is just a shortcut to not re-write every time the access to the sols_per_product and to make things more readable. Since i got rid of the ugly [res[0].id] that used to be there, we can get rid of that temp variable too.